language_service/src/compilation.rs

```diff
@@ -123,6 +124,17 @@ impl Compilation {
 
         let (package_store, package_id) = compiler.into_package_store();
 
+        // Compute new lints and append them to the errors Vec.
```

Comment:
We should only execute this bit if `errors` is empty (i.e. no parse, resolve, typecheck, pass, etc errors). 

For performance, among other reasons: we don't want to waste time running lints every few keystrokes, if the user is in the middle of typing a statement, for example.